File: BaekjoonOnlineJudge/9012/9012.py
import sys

input = sys.stdin.readline

def isCorrect(parens):
    stack = []

    for paren in parens:
        if not stack:
            stack.append(paren)
        else:
            if paren == ")":
                if not stack or stack[-1] != "(":
                    return "NO"
                else:
                    stack.pop()
            else:
                stack.append(paren)
    
    if stack:
        return "NO"
    else:
        return "YES"



T = int(input())

for _ in range(T):
    parens = list(input().rstrip())
    
    print(isCorrect(parens))

File: BaekjoonOnlineJudge/9012/9012.java
import java.io.*;
import java.util.*;


public class Main {

    private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    private static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    private static StringTokenizer st;


    public static void main(String[] args) throws IOException {
        int n = Integer.parseInt(br.readLine());

        while (n-- != 0) {
            Stack<Character> stack = new Stack<>();
            boolean isPossible = true;

            String input = br.readLine();

            for (int i = 0; i < input.length(); i++) {
                char ps = input.charAt(i);

                if (ps == '(') {
                    stack.add(ps);
                } else {
                    if (!stack.empty() && stack.peek() == '(') {
                        stack.pop();
                    } else {
                        isPossible = false;
                        break;
                    }
                }
            }

            if (isPossible && stack.empty()) {
                bw.write("YES\n");
            } else {
                bw.write("NO\n");
            }
        }

        bw.flush();
        bw.close();
        br.close();
    }
}